yt_dlp/YoutubeDL.py

```diff
@@ -2810,10 +2810,14 @@ def process_subtitles(self, video_id, normal_subtitles, automatic_captions):
                     self.params.get('subtitleslangs'), {'all': all_sub_langs}, use_regex=True)
             except re.error as e:
                 raise ValueError(f'Wrong regex for subtitlelangs: {e.pattern}')
-        elif normal_sub_langs:
-            requested_langs = ['en'] if 'en' in normal_sub_langs else normal_sub_langs[:1]
         else:
-            requested_langs = ['en'] if 'en' in all_sub_langs else all_sub_langs[:1]
+            # Check for `en` subs first
+            subs_list = sorted(filter(lambda f: f.startswith('en'), normal_sub_langs))
+            subs_list.extend(sorted(filter(lambda f: f.startswith('en'), all_sub_langs)))
+            # Append all langs at the end in case there are no `en` subs
+            subs_list.extend(sorted(normal_sub_langs))
+            subs_list.extend(sorted(all_sub_langs))
+            requested_langs = subs_list[:1]
```

Comment:
(untested)

```suggestion
            requested_langs = LazyList(itertools.chain(
                ['en'] if 'en' in normal_sub_langs else [],
                filter(lambda f: f.startswith('en'), normal_sub_langs),
                ['en'] if 'en' in all_sub_langs else [],
                filter(lambda f: f.startswith('en'), all_sub_langs)),
                normal_sub_langs, all_sub_langs,
            )[:1]
```

Other than code cleanup, this changes:
* Prefer exact `en` if available - backward compat, avoid defaulting CC etc
* Don't re-sort - Except for the `en` prioritization, it's supposed to use order given by IE. Why did u add sorting?

Reply:
Okay thanks! Your approach is more explicit and nicer. 

> Why did u add sorting?

It was to automatically put `en` first and then any `en-*`, which basically avoids two of the explicit 'en` checks in the new code.